articles/tokens/concepts/token-storage.md

```diff
@@ -0,0 +1,102 @@
+- [Saving and Renewing Tokens for Swift](/libraries/auth0-swift/save-and-refresh-jwt-tokens)
+- [Native/Mobile Apps Quickstarts](/quickstart/native)
+
+## Single-page app scenarios
+
+When the SPA calls only an API that is served from a domain that can share cookies with the domain of the SPA, no tokens are needed. OAuth adds additional attack vectors without providing any additional value and should be avoided in favor of a traditional cookie-based approach. For an overview of this approach and an example implementation, see [Single-Page App Authentication Using Cookies](/login/spa/authenticate-with-cookies).
+
+When the SPA calls multiple APIs that reside in a different domain, access and optionally refresh tokens are needed.
+
+-  If the SPA backend can handle the API calls, handle tokens server-side using:
+    - [Authorization Code Flow](/flows/concepts/auth-code)
+    - [Authorization Code Flow with Proof Key for Code Exchange (PKCE)](/flows/concepts/auth-code-pkce)
+    - [Hybrid Flow](/api-auth/grant/hybrid)
+
+- If the SPA backend cannot handle the API calls, the tokens should be stored in the SPA backend but the SPA needs to fetch the tokens from the backend to perform requests to the API. A protocol needs to be established between the backend and the SPA to allow the secure transfer of the token from the backend to the SPA.
+
+- If you have a SPA with **no** corresponding backend server, your SPA should request new tokens on login and store them in memory without any persistence. To make API calls, your SPA would then use the in-memory copy of the token.
+
+### Browser in-memory scenarios
+
+Auth0 recommends storing tokens in browser memory as the most secure option. Using [Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) to handle the transmission and storage of tokens is the best way to protect the tokens, as Web Workers run in a separate global scope than the rest of the application. Use [auth0-spa-js](https://github.com/auth0/auth0-spa-js) whose default storage option is in-memory storage leveraging Web Workers.
+
+If you cannot use Web Workers, Auth0 recommends as an alternative that you use [JavaScript closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#Emulating_private_methods_with_closures) to emulate private methods.
+
+::: warning
+The in-memory method for browser storage **does not** provide persistence across page refreshes and browser tabs. 
+:::
+
+### Browser local storage scenarios
+
+Storing tokens in browser local storage provides persistence across page refreshes and browser tabs, however if an attacker can achieve running JavaScript in the SPA, they can retrieve the tokens stored in local storage using a cross-site scripting (XSS) attack. A vulnerability leading to a successful XSS attack can be either in the SPA source code or in any third-party JavaScript code (such as bootstrap, jQuery, or Google Analytics) included in the SPA.
```

Comment:
```suggestion
Storing tokens in browser local storage provides persistence across page refreshes and browser tabs, however if an attacker can achieve running JavaScript in the SPA using a cross-site scripting (XSS) attack, they can retrieve the tokens stored in local storage. A vulnerability leading to a successful XSS attack can be either in the SPA source code or in any third-party JavaScript code (such as bootstrap, jQuery, or Google Analytics) included in the SPA.
```

Reply:
I added this manually - as I had done it already based on your Slack comment.